Extend `shouldDisableFullscreenVideoAspectRatioAdaptiveSizing` quirk to `x.com`

Ahmad-S792 opened the pull request:
#### dd4682ec1201f33ac774c80ead6c4c37622e8ce3
<pre>
Extend `shouldDisableFullscreenVideoAspectRatioAdaptiveSizing` quirk to `x.com`
<a href="https://bugs.webkit.org/show_bug.cgi?id=282606">https://bugs.webkit.org/show_bug.cgi?id=282606</a>
<a href="https://rdar.apple.com/132850672">rdar://132850672</a>

Reviewed by Brent Fulgham.

This is still an issue on visionOS for certain videos, so extending the
quirk to cover `x.com`.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldDisableFullscreenVideoAspectRatioAdaptiveSizing const):

Canonical link: <a href="https://commits.webkit.org/286189@main">https://commits.webkit.org/286189@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f953bdaefa682b0c7ea14a09843fc4bb1feb6047

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/75058 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/54495 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/27889 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/79505 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/26309 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/77175 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/63632 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/2274 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/58925 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/17188 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/78125 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/49087 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/64486 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/39303 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/46436 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/21990 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/24635 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/67538 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/22330 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/80986 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/2383 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/1464 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/67181 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/2533 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/64498 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/66473 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16538 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/10411 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/8588 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/2347 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/5147 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/2371 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/3297 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/2378 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -1411,13 +1411,13 @@ bool Quirks::allowLayeredFullscreenVideos() const
         return false;
 
-    return isDomain("twitter.com"_s);
+    return isDomain("x.com"_s);
```
Please add a comment:
// FIXME(rdar://124579556): Remove once x.com adjusts video handling for visionOS

Ahmad-S792 replied:
@brentfulgham - Done. :-)

brentfulgham reviewed (approved):
Please add a comment about the Evangelism task so we remember to remove this in the future.

karlcow reviewed (approved):
lgtm

Ahmad-S792 committed:
Extend `shouldDisableFullscreenVideoAspectRatioAdaptiveSizing` quirk to `x.com`
https://bugs.webkit.org/show_bug.cgi?id=282606
rdar://132850672

Reviewed by Brent Fulgham.

This is still an issue on visionOS for certain videos, so extending the
quirk to cover `x.com`.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldDisableFullscreenVideoAspectRatioAdaptiveSizing const):

Canonical link: https://commits.webkit.org/286189@main
